### PR TITLE
feat: styles operation header

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksoperation-header-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1204678746647847/1205079140949241/f) 🎟️

## 🗒️ What

Styles out the operation header component. Updates the server file to truncate the path.

## 📸 Design Screenshots

Screenshot 2023-07-25 at 12 01 08 PM

## 🧪 Testing

- [Visit the page](https://dev-portal-git-ksoperation-header-hashicorp.vercel.app/hcp/api-docs/vault-secrets)
- Validate that is [matches the designs](https://www.figma.com/file/wjoC3ZVsA1kDuPsFLjEFuY/API-Template?type=design&node-id=526-34648&mode=design&t=T8TGJZC7PM2lxZrj-0)

## 💭 Anything else?

Note there aren't mobile specific designs yet for this component. So this reflects the desktop spec, although shouldn't be 'broken' at mobile. Mostly just missing some global layout padding which should be handled elsewhere.

### CODE REVIEW

File: config/preview.json
Comment (kendallstrautman) [inline config/preview.json:7]: TODO - don't merge until this is flipped back to `false`

File: src/views/open-api-docs-view/utils/get-operation-props.ts
Comment (kendallstrautman) [inline src/views/open-api-docs-view/utils/get-operation-props.ts:70]: @zchsh Wasn't sure here how you wanted to handle adding in the props. I figured it'd be nice to align the component props with the `OperationProps`. Let me know!

**Comment by braydenlove** [comment]
This looks great @kendallstrautman. Can we make sure the `h4` is `foreground-strong`? Currently it appears to be #000000. Otherwise this is good to go.

File: src/views/open-api-docs-view/components/open-api-operations/index.tsx
Comment (zchsh) [inline src/views/open-api-docs-view/components/open-api-operations/index.tsx:46]: ⛏️ Okay this is a weird one, and I think is also a problem upstream, but long `path` and also long `headerText` strings can cause overflow and wreak havok on the layout on small viewports, cause they tend to not have any spaces or word-break-able characters:

| overflow issue |
| - |
| overflow-issue |

A "quick fix" could be `overflow-wrap: break-word`, I think 🤔 A more involved fix might be to get `<wbr>` elements interspersed into `headerText` and `path` somehow, which might yield a more polished result:

| `overflow-wrap` | `<wbr>` |
| - | - |
| overflow-wrap-anywhere | injected-word-breaks |

I think we'll actually want `overflow-wrap: break-word` even if we go for the more involved `<wbr>` addition, since we still won't be able to guarantee we've added `<wbr>` in enough spots... so that's something at least! I'd definitely say we could roll with that CSS only fix for now 👍

---

... that being said but I can't help jumping to the conclusion that we might want to pursue the `<wbr>` approach, so I noodled on that over in a branch off this PR:

- https://github.com/hashicorp/dev-portal/pull/2103

File: src/views/open-api-docs-view/components/operation-header/operation-header.module.css
Comment (zchsh) [inline src/views/open-api-docs-view/components/operation-header/operation-header.module.css:10]: Quick fix for overflow issues on mobile:

```suggestion
	font-weight: var(--token-typography-font-weight-bold);
	overflow-wrap: break-word;
```

File: src/views/open-api-docs-view/components/operation-header/operation-header.module.css
Comment (zchsh) [inline src/views/open-api-docs-view/components/operation-header/operation-header.module.css:21]: Quick fix for overflow issues on mobile:

```suggestion
	color: var(--token-color-foreground-primary);
	overflow-wrap: break-word;
```

**Review by zchsh — APPROVED** [review]
✅ ✨ Looking great!

🚧 One almost-blocking thing is the `overflow-wrap` issue. I added some suggestions to add `overflow-wrap: break-word`, so approving pending those additions!

💭 Very optionally, if you wanna add `<wbr >` inject to this PR you could grab or merge in the work from #2103 , but could very much see the case to make that a separate PR entirely rather than roll it in here 👍

File: src/views/open-api-docs-view/components/open-api-operations/index.tsx
Comment (kendallstrautman) [reply src/views/open-api-docs-view/components/open-api-operations/index.tsx:46]: I love your exploration with the word-breaks and the idea that we could life that into a higher utility. I'll push up the CSS only fix for now so we can play around more with the work break approach.

**Comment by kendallstrautman** [comment]
@braydenlove updated that color! Thanks for catching that.

**Pull request merged**
